### ISSUE TITLE
Page context error after login

### ISSUE THREAD

**Issue by RussellEngland** [catalyst/moodle-auth_saml2#151]
Hey Chaps

Great plugin! :)

I get this error after logging in, with debugging set to developer, create user if it doesn't exist and mapping to a checkbox custom field.

I think the page context is required by the menu custom field type? Could be a Totara error rather than a plugin error?

This was using version "2017081700" of the plugin - last commit was 5551eb7376f4365596979d6c01ad5b4d2e211556

Tested with Totara 2.7.6

`Coding problem: $PAGE->context was not set. You may have forgotten to call require_login() or $PAGE->set_context(). The page may not display correctly as a result
line 458 of /lib/pagelib.php: call to debugging()
line 783 of /lib/pagelib.php: call to moodle_page->magic_get_context()
line 1319 of /lib/weblib.php: call to moodle_page->__get()
line 62 of /user/profile/field/menu/field.class.php: call to format_string()
line 596 of /user/profile/lib.php: call to profile_field_menu->profile_field_menu()
line 3785 of /lib/moodlelib.php: call to profile_save_data()
line 306 of /auth/saml2/auth.php: call to create_user_record()
line 32 of /auth/saml2/login.php: call to auth_plugin_saml2->saml_login()`

**Comment by brendanheywood** [comment]
Yes I think this may be the same as this issue: https://github.com/catalyst/moodle-auth_saml2/issues/103

A fix was added, but then revoke as it caused wider issues so was revoked. The proper fix will probably end up quite simple but probably need a few hours to resolve.

It would also help diagnosis is if you could retest it with this function commented out:

https://github.com/catalyst/moodle-auth_saml2/blob/MOODLE_UPTO32/auth.php#L177-L193

If that works then you could leave it like that as a workaround until a proper fix can be sorted. As always pull requests are welcome, but take care and read that other issue first as it's not trivial.
